tests/test_change_plan.py

```diff
@@ -0,0 +1,159 @@
+    is_kopf_handler_finished,
+    start_cluster,
+)
+
+
+async def do_pods_exist(core: CoreV1Api, namespace: str, expected: Set[str]) -> bool:
+    pods = await core.list_namespaced_pod(namespace=namespace)
+    return expected.issubset({p.metadata.name for p in pods.items})
+
+
+async def do_pod_ids_exist(core: CoreV1Api, namespace: str, pod_ids: Set[str]) -> bool:
+    pods = await core.list_namespaced_pod(namespace=namespace)
+    return bool(pod_ids.intersection({p.metadata.uid for p in pods.items}))
+
+
+@pytest.mark.k8s
+@pytest.mark.asyncio
+async def test_up_cpu_and_ram(faker, namespace, kopf_runner, api_client):
+    coapi = CustomObjectsApi(api_client)
+    core = CoreV1Api(api_client)
+    name = faker.domain_word()
+    # start_cluster creates a cluster with cpu=2 and memory=4Gi per node
+    host, password = await start_cluster(name, namespace, core, coapi, 3, "5.0.0")
+
+    await assert_wait_for(
+        True,
+        do_pods_exist,
+        core,
+        namespace.metadata.name,
+        {
+            f"crate-data-hot-{name}-0",
+            f"crate-data-hot-{name}-1",
+            f"crate-data-hot-{name}-2",
+        },
+    )
+
+    conn_factory = connection_factory(host, password)
+
+    await assert_wait_for(
+        True,
+        is_cluster_healthy,
+        conn_factory,
+        3,
+        err_msg="Cluster wasn't healthy",
+        timeout=DEFAULT_TIMEOUT,
+    )
+
+    await create_test_sys_jobs_table(conn_factory)
+
+    pods = await core.list_namespaced_pod(namespace=namespace.metadata.name)
+    original_pods = {p.metadata.uid for p in pods.items}
+
+    await coapi.patch_namespaced_custom_object(
+        group=API_GROUP,
+        version="v1",
+        plural=RESOURCE_CRATEDB,
+        namespace=namespace.metadata.name,
+        name=name,
+        body=[
+            {
+                "op": "replace",
+                "path": "/spec/nodes/data/0/resources/limits/cpu",
+                "value": 1,  # Increase from 2
+            },
+            {
+                "op": "replace",
+                "path": "/spec/nodes/data/0/resources/limits/memory",
+                "value": "5Gi",  # Increase from 4Gi
+            },
+        ],
+    )
+
+    await assert_wait_for(
+        True,
+        cluster_routing_allocation_enable_equals,
+        connection_factory(host, password),
+        "new_primaries",
+        err_msg="Cluster routing allocation setting has not been updated",
+        timeout=DEFAULT_TIMEOUT * 5,
+    )
+
+    await assert_wait_for(
+        True,
+        is_kopf_handler_finished,
+        coapi,
+        name,
+        namespace.metadata.name,
+        "operator.cloud.crate.io/cluster_update.change_plan",
+        err_msg="Plan change has not finished",
+        timeout=DEFAULT_TIMEOUT * 5,
+    )
+
+    await assert_wait_for(
+        False,
+        do_pod_ids_exist,
+        core,
+        namespace.metadata.name,
+        original_pods,
+        timeout=DEFAULT_TIMEOUT * 15,
+    )
+
```

Comment:
You might also want to assert that the limits/requests have actually changed on the new pods.